test/evergreen/evg_cfg.py

```diff
@@ -0,0 +1,335 @@
+#! /usr/bin/env python3
+
+"""
+This program provides an CLI interface to check and generate Evergreen configuration.
+"""
+
+import os
+import sys
+import re
+import subprocess
+import docopt
+
+TEST_TYPES = ('make_check', 'csuite')
+EVG_CFG_FILE = "test/evergreen.yml"
+MAKE_SUBDIRS_FILE = "build_posix/Make.subdirs"
+CSUITE_TEST_DIR = "test/csuite"
+MAKE_CHECK_TEST_TMPLT = "test/evergreen/make_check_test_evg_task.template"
+CSUITE_TEST_TMPLT = "test/evergreen/csuite_test_evg_task.template"
+MAKE_CHECK_TEST_SEARCH_STR = "  # End of normal make check test tasks"
+CSUITE_TEST_SEARCH_STR = "  # End of csuite test tasks"
+WIREDTIGER_REPO = "git@github.com:wiredtiger/wiredtiger.git"
+
+# This list of sub directories will be skipped from checking
+# They are not expected to trigger any 'make check' testing.
+make_check_subdir_skips = [
+    "ext/collators/reverse",
+    "ext/collators/revint",
+    "ext/compressors/lz4",
+    "ext/compressors/nop",
+    "ext/compressors/snappy",
+    "ext/compressors/zlib",
+    "ext/compressors/zstd",
+    "ext/datasources/helium",
+    "ext/encryptors/nop",
+    "ext/encryptors/rotn",
+    "ext/extractors/csv",
+    "ext/test/kvs_bdb",
+    "ext/test/fail_fs",
+    "api/leveldb",
+    "lang/java",
+    "test/utility",
+    "examples/java",
+    "test/csuite",  # csuite has its own set of Evergreen tasks, skip the checking here
+    "test/syscall",
+    "bench/workgen",
```

Comment:
This list is long enough I think it should be sorted.

Reply:
Ack.